docs/en/userguide/advanced_setup.md

```diff
@@ -0,0 +1,277 @@
+
+Additional information can be stored like page locale (relevant for `Fluent` module) to specify the target data even further.
+
+### Unit tests
+
+Writing unit tests for `Elemental` with `Fluent` can be rather tricky to figure out.
+Here are some guidelines to make that easier.
+
+#### Make sure your fixture has some locales setup
+
+It's important to include some locales because otherwise your test might be testing a very different situation.
+Example `Locale` setup in a fixture:
+
+```
+TractorCow\Fluent\Model\Locale:
+  nz:
+    Locale: en_NZ
+    Title: 'English (New Zealand)'
+    URLSegment: newzealand
+  us:
+    Locale: en_US
+    Title: 'English (US)'
+    URLSegment: usa
+    Fallbacks:
+      - =>TractorCow\Fluent\Model\Locale.nz
+```
+
+#### Localised fixture data (automatic, single locale)
+
+In the case your fixture needs to contain data for only a single locale you can specify your desired locale in your unit test like this:
+
+```
+protected function setUp()
+{
+    // Set locale for fixture creation
+    FluentState::singleton()->withState(function (FluentState $state): void {
+        $state->setLocale('en_NZ');
+        parent::setUp();
+    });
+}
+
+```
+
+This will localise all your data so you don't need to worry about that in your fixtures. Following fixture will produce a page localised in `en_NZ`:
+
+```
+App\Pages\OperatorArticlePage:
+  article-page1:
+    Title: ArticlePage1 NZ
+    URLSegment: article-page1
+```
+
+
+
+#### Localised fixture data (manual, single or multiple locales)
+
+In some cases you want to have multiple locales of one page set up in your fixtures.
+This means you need to specify localised data manually.
+Example below shows how to specify localised `Title` for two locales of one page.
+Note that each localised field has to be specified for the table that actually holds the field.
+In this case, it's `SiteTree`.
+If you are unsure where your field sits it may be a good idea to check your database structure first and find the relevant table.
+
+```
+App\Pages\OperatorArticlePage:
+  article-page1:
+    Title: ArticlePage1
+    URLSegment: article-page1
+
+SiteTree_Localised:
+  article-page1-nz:
+    Locale: en_NZ
+    RecordID: =>App\Pages\OperatorArticlePage.article-page1
+    Title: ArticlePage1 NZ
+  article-page1-au:
+    Locale: en_AU
+    RecordID: =>App\Pages\OperatorArticlePage.article-page1
+    Title: ArticlePage1 AU
+```
+
+### Working with Fluent state
+
+Make sure you always use the `FluentState` callback to change the global state like this:
+
+```
+FluentState::singleton()->withState(function (FluentState $state) {
+    $state->setLocale('en_NZ');
+
+    // your code goes here
+})
+```
+
+This is very important as global state is reverted back after the callback is executed so it's safe to be used.
+Unit tests benefit mostly from this as this makes sure that there are no dependencies between unit tests as the global state is always changed only locally in one test.
```

Comment:
```suggestion
Since this makes sure that there are no dependencies between unit tests as the global state is always changed only locally in one test, unit tests benefit mostly from using FluentState callbacks.
```